tests/integ/modin/binary/test_binary_op.py

```diff
@@ -2586,3 +2586,19 @@ def test_df_sub_series():
     eval_snowpark_pandas_result(
         snow_df, native_df, lambda df: df.sub(df["two"], axis="index"), inplace=True
     )
+
+
+@sql_count_checker(query_count=1, join_count=0)
+def test_binary_op_series_from_same_df():
```

Comment:
The test seems only have 1 align, this test more than 1 align, because typically the first align can already be optimized away without the optimization. let me add some comment here